File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginModule.java
package com.charlye934.loginddagger.login;

import dagger.Module;
import dagger.Provides;

@Module
public class LoginModule {
    @Provides
    public LoginActivityMVP.Presenter provideLoginActivityPresenter(LoginActivityMVP.Model model){
        return new LoginActivityPresenter(model);
    }

    @Provides
    public LoginActivityMVP.Model provideLginActivityModel(LoginRepository repository){
        return new LoginActivityModel(repository);
    }

    @Provides
    public LoginRepository provideLoginRepository(){
        return new  MemoryRepository();
    }

}

File: RxJavaExamples/RxJavaDagger/LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginModule.kt
package com.charlye934.loginddagger.login

import dagger.Module
import dagger.Provides

@Module
class LoginModule {

    @Provides
    fun provideLoginActivityPresenter(
            model: LoginActivityMVP.Model
    ): LoginActivityMVP.Presenter{
        return LoginActivityPresenter(model)
    }

    @Provides
    fun provideLoginActivityModel(
            repository: LoginRepository
    ): LoginActivityMVP.Model{
        return LoginActivityModel(repository)
    }

    @Provides
    fun provideLoginRepository(): LoginRepository{
        return MemoryRepository()
    }
}